src-java/connecteddevices-topology/connecteddevices-storm-topology/src/main/java/org/openkilda/wfm/topology/connecteddevices/service/PacketService.java

```diff
@@ -97,8 +97,6 @@ public void handleLldpData(LldpInfoData data) {
             device.setTimeLastSeen(Instant.ofEpochMilli(data.getTimestamp()));
             device.setFlowId(flowRelatedData.flowId);
             device.setSource(flowRelatedData.source);
-
-            switchConnectedDeviceRepository.createOrUpdate(device);
```

Comment:
I'm not very familiar with new DB but how it should work? After changing local Instance of `device` it will be saved in DB automatically?

Reply:
The implementation relies on the concept of persistence context ([persistence-layer](https://github.com/telstra/open-kilda/blob/451f4ef5125bd0eda224df872c5f5d1b2a949744/docs/design/data-persistence/persistence-layer.md)). When the entity instance is in the persistent state, all changes that you make to this instance will be applied to the corresponding storage records upon committing of the transaction or closing the persistence context. This means that you don't have to call save, update or other repository methods to get these changes to the storage.
The same approach is used in JPA, Hibernate and many other solutions.